lib/td/client/api.rb

```diff
@@ -272,27 +272,46 @@ def do_get(url, params=nil, opt={}, &block)
     response = nil
     etag = nil
     current_total_chunk_size = 0
+    body = String.new unless block_given?
     begin # this block is to allow retry (redo) in the begin part of the begin-rescue block
       begin
         if etag
           header['If-Range'] = etag
           header['Range'] = "bytes=#{current_total_chunk_size}-"
         else
+          etag = nil
           current_total_chunk_size = 0
+          body.clear if body
         end
-        if block
+
+        if block_given?
           response = client.get(target, params, header) {|res, chunk|
             current_total_chunk_size += chunk.bytesize
-            block.call(res, chunk, current_total_chunk_size)
+            yield res, chunk, current_total_chunk_size
           }
         else
           response = client.get(target, params, header)
           current_total_chunk_size += response.body.bytesize
+          body << response.body
         end
+
         # XXX ext/openssl raises EOFError in case where underlying connection causes an error,
         #     but httpclient ignores it. Therefor, check content size.
         #     https://github.com/nahi/httpclient/issues/296
-        validate_content_length!(response, current_total_chunk_size)
+        if expected_size = response.header['Content-Range'].first
+          expected_size = expected_size[/\d+$/]
+        else
+          expected_size = response.header['Content-Length'].first
+        end
+        if expected_size
+          expected_size = expected_size.to_i
+          if expected_size != current_total_chunk_size
+            if expected_size < current_total_chunk_size
+              etag = false
```

Comment:
> But I would think resuming is needed during server is unhealthy. So the server easily returns 5xx when resuming.

Resuming is for the case network is unstable.
If the server is unhealthy the result can't be trusted.

Anyway for `5xx` the exception is not rescued.